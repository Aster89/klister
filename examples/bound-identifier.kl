#lang kernel

(import (shift kernel 1))

(define-macros
  ((m (lambda (stx)
        (syntax-case stx
          ((list (_ x y))
           (>>= (bound-identifier=? x y)
                (lambda (bool)
                  (pure (if bool ''t ''f))))))))))

(example (m x x))
(example (m x y))
