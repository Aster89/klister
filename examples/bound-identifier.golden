#[bound-identifier.kl:11.36-11.37]<t> : Syntax
#[bound-identifier.kl:11.40-11.41]<f> : Syntax
